src/AvailabilityManager.php

```diff
@@ -32,16 +36,33 @@ public function getCheckers() {
    * {@inheritdoc}
    */
   public function check(PurchasableEntityInterface $entity, $quantity, Context $context) {
+    $min = $max = 0;
+
+    $has_opinion = FALSE;
     foreach ($this->checkers as $checker) {
       if ($checker->applies($entity)) {
-        $result = $checker->check($entity, $quantity, $context);
-        if ($result === FALSE) {
-          return FALSE;
+        $response = $checker->check($entity, $quantity, $context);
+        if ($response instanceof AvailabilityResponseUnavailable) {
+          return $response;
         }
+        if ($response instanceof AvailabilityResponseNeutral) {
+          continue;
+        }
+        $has_opinion = TRUE;
```

Comment:
I think this should be reversed
**$min = max($min, $response->getMin());
$max = min($max, $response->getMax());**
As its I think it in consistent with:
**if ($response instanceof AvailabilityResponseUnavailable) {
          return $response;** 
i.e. it is the interaction of the availability checkers.  

Real life example:
stock module says: max: 500
Max per order module says: max: 10 (per order)
In such case I think the maximum should be the min of the two: 10